### PR TITLE
1659 - Add support for custom.js file

### PR DESCRIPTION
**Pull request by cshyam1892**

### CODE REVIEW

**Comment by cshyam1892** [comment]
Thanks for your feedback. 

Would it be possible for you to take a look at some changes I made?

Thanks a lot :)

**Comment by cshyam1892** [comment]
Ahh..I see what you mean now, thanks for that!

So what will be the right approach without inline script?

Do we add an extra option to add script-src?

> You can go on with a similar approach if you add a new config option only for JS code.

How would this work? How would external scripts be inserted and executed in a new config option?

Thanks for putting up with me.

**Comment by ildyria** [comment]
While I appreciate the intent, I disagree with the step suggested.
I rather prefer the following approach https://github.com/LycheeOrg/Lychee/pull/1697 which is much more sane as it requires server side file modification to execute specific JS instead of DB compromise.

**Comment by cshyam1892** [comment]
> While I appreciate the intent, I disagree with the step suggested. I rather prefer the following approach [LycheeOrg/Lychee#1697](https://github.com/LycheeOrg/Lychee/pull/1697) which is much more sane as it requires server side file modification to execute specific JS instead of DB compromise.

Thank you so much for the reply.

Yeah, that makes sense. I would rather create a new config option in the settings for the custom js then add js directly from the footer text section. However, the documents need to be updated.

https://lycheeorg.github.io/docs/faq_general.html#how-to-add-custom-scripts

**Comment by ildyria** [comment]
You are looking for something that is wayy too complex for what it should be. :')

What I meant by 
> I rather prefer the following approach https://github.com/LycheeOrg/Lychee/pull/1697 which is much more sane as it requires server side file modification to execute specific JS instead of DB compromise.

Is that you can just follow the same approach as for `user.css`. and provide a place holder custom.js file just like we do for `user.css`, if a user wants to modify it then it can easilly do so.

As a result you can literally just ignore the Lychee-front part.

TL;DR: That line is enough
https://github.com/LycheeOrg/Lychee/pull/1697/files#diff-63c3564a7487f1e735e12235ec80fefc998b4ad08b4753c56db0ef6dc6731c4fR38

Combine with a bit of php here:
https://github.com/LycheeOrg/Lychee/blob/master/scripts/install_files.php#L14

You have a working solution that is significantly ligher. :+1:

File: scripts/main/view.js [hunk 1]
```diff
@@ -1676,12 +1677,34 @@ view.settings = {
...
+			<p>${lychee.locale["JS_TEXT"]}</p>
+			<textarea id="js"></textarea>
+			<div class="basicModal__buttons">
+				<a id="basicModal__action_set_js" class="basicModal__button">${lychee.locale["JS_TITLE"]}</a>
+			</div>
+			</div>`;
+
+			$(".settings_view").append(msg);
+
+			api.getJS("dist/custom.js", function (data) {
```
[the remark this answers is not in the document]
Comment (cshyam1892) [reply scripts/main/view.js]: done :)

File: scripts/main/lychee_locale.js [hunk 2]
```diff
@@ -378,6 +379,8 @@ lychee.locale = {
 
 	CSS_TEXT: "Personalize CSS:",
 	CSS_TITLE: "Change CSS",
+	JS_TEXT: "Custom JS:",
```
[the remark this answers is not in the document]
Comment (cshyam1892) [reply scripts/main/lychee_locale.js:382]: Updated only for English.

File: scripts/main/view.js [hunk 3]
```diff
@@ -1676,12 +1677,36 @@ view.settings = {
...
+			<p>${lychee.locale["JS_TEXT"]}</p>
+			<textarea id="js"></textarea>
+			<div class="basicModal__buttons">
+				<a id="basicModal__action_set_js" class="basicModal__button">${lychee.locale["JS_TITLE"]}</a>
+			</div>
+			</div>`;
+
+			$(".settings_view").append(msg);
+
+			let js_addr = $($("script")[5]).attr("src");
```
[the remark this answers is not in the document]
Comment (cshyam1892) [reply scripts/main/view.js]: done :)

File: scripts/api.js [hunk 4]
```diff
@@ -183,3 +183,43 @@ api.getCSS = function (url, callback) {
...
 	});
 };
+
+/**
+ *
+ * @param {string} url
+ * @param {APISuccessCB} callback
+ * @returns {void}
+ */
+api.getJS = function (url, callback) {
```
Comment (ildyria) [inline scripts/api.js:193]: Isn't this function a literal copy past of `api.getCSS` ? 
If so, then this is duplicate code, you can just rename `getCSS` into something like `getRawContent` and replace at the appropriate places in the code and it should work pretty nicely.
